File: tests/fixtures/edgecase_repo/python/pkg/module3.py
"""Module 3 of the generated Python layer."""

from __future__ import annotations

from pkg.module2 import Service2


class Base3:
    """Base class for service 3."""


class Service3(Base3):
    """Service 3."""

    def handle(self, payload):
        """Handle a payload."""
        self.validate(payload)
        return Service2().handle(payload)

    def validate(self, payload):
        """Validate a payload."""
        return payload is not None


def helper_3(value):
    """Module-level helper."""
    return value

File: tests/fixtures/edgecase_repo/dart/module3.dart
import 'dart:async';

import 'module2.dart';

abstract class Handler3 {
  bool handle(String payload);
}

class Base3 {
  final String name;
  Base3(this.name);
}

class Service3 extends Base3 implements Handler3 {
  Service3(String name) : super(name);

  @override
  bool handle(String payload) {
    this.validate(payload);
    return true;
  }

  void validate(String payload) {}
}

String helper3(String value) => value;
